### PR TITLE
4783 bug model comparison without model cards should add model specific observations in the table

### PR DESCRIPTION
**Pull request by dgauldie**
# Description

* Include a summary of the changes and the related issue. 
* Include relevant motivation and context.

Resolves #(issue)

### CODE REVIEW

File: packages/client/hmi-client/src/components/workflow/ops/model-comparison/tera-model-comparison.vue [hunk 1]
```diff
@@ -287,9 +287,47 @@ async function buildJupyterContext() {
...
+	return Object.values(obj).some((value) => {
+		if (Array.isArray(value)) {
+			return value.length > 0;
+		}
+		if (typeof value === 'object' && value !== null) {
+			return Object.keys(value).length > 0;
+		}
+		return value !== null && value !== undefined && value !== '';
+	});
+}
```
Comment (YohannParis) [inline packages/client/hmi-client/src/components/workflow/ops/model-comparison/tera-model-comparison.vue:300]: You can use [Lodash isEmtpy](https://lodash.com/docs/4.17.15#isEmpty)

```suggestion
function hasNonEmptyValue(obj) {
	return Object.values(obj).some((value) => !isEmpty(value));
}
```

File: packages/client/hmi-client/src/components/workflow/ops/model-comparison/tera-model-comparison.vue [hunk 2]
```diff
@@ -287,9 +287,47 @@ async function buildJupyterContext() {
...
+		markdown += mdi.render(`## Metadata Comparisons:\n\n`);
+		markdown += mdi.render(`### Details:\n`);
+		markdown += mdi.render(`${comparison.metadataComparison.description}\n\n`);
+		markdown += mdi.render(`### Uses:\n`);
+		markdown += mdi.render(`${comparison.metadataComparison.uses}\n\n`);
+		markdown += mdi.render(`### Bias, Risks, and Limitations:\n`);
+		markdown += mdi.render(`${comparison.metadataComparison.biasRisksLimitations}\n\n`);
+		markdown += mdi.render(`### Testing and Validation:\n`);
+		markdown += mdi.render(`${comparison.metadataComparison.testing}\n\n`);
+	}
```
Comment (YohannParis) [inline packages/client/hmi-client/src/components/workflow/ops/model-comparison/tera-model-comparison.vue:329]: Why not render only once using template literal?
```suggestion
const mdi = markdownit()
const sections = `
# ${comparison.title}

## Summary:
${comparison.summary}

## Structural Comparisons:

### States
${comparison.semanticComparison.states}

### Transitions
${comparison.semanticComparison.transitions}

### Parameters:
${comparison.semanticComparison.parameters}

### Observables:
${comparison.semanticComparison.observables}
`;

markdown += sections;

if (!isEmpty(comparison.metadataComparison)) {
	const metadataSections = `
## Metadata Comparisons:

### Details:
${comparison.metadataComparison.description}

### Uses:
${comparison.metadataComparison.uses}

### Bias, Risks, and Limitations:
${comparison.metadataComparison.biasRisksLimitations}

### Testing and Validation:
${comparison.metadataComparison.testing}
`;
	markdown += metadataSections;
}

markdown = mdi.render(markdown);
```

**Review by YohannParis — APPROVED** [review]
LGTM

File: packages/client/hmi-client/src/components/workflow/ops/model-comparison/tera-model-comparison.vue (hunk 1 above)
Comment (dgauldie) [reply packages/client/hmi-client/src/components/workflow/ops/model-comparison/tera-model-comparison.vue]: fixed

File: packages/client/hmi-client/src/components/workflow/ops/model-comparison/tera-model-comparison.vue (hunk 2 above)
Comment (dgauldie) [reply packages/client/hmi-client/src/components/workflow/ops/model-comparison/tera-model-comparison.vue]: fixed

**Review by shawnyama — APPROVED** [review]

**Pull request merged**